tests/python/unittest/test_operator.py

```diff
@@ -1581,6 +1581,27 @@ def test_bmin(a, b):
         data = gen_broadcast_data(idx=200)
         check_bmaxmin_gradient(c, data[0], data[1], 0.001, 1e-2, 1e-3)
 
+    def test_band(a, b):
+        c = mx.sym.broadcast_logical_and(a, b)
+        check_binary_op_forward(c, lambda x, y: np.logical_and(x, y), gen_broadcast_data, mx_nd_func=mx.nd.broadcast_logical_and)
```

Comment:
Why aren't there also non-broadcast versions (which will be much faster) and then you can just name it logical_or and use broadcast in the python when the shapes don't match similar to NDArray add?  Seems cumbersome that the numpy will be logical_or and for mxnet, you have to remember it's broadcast_logical_or

Reply:
The broadcast_xxx function actually wraps the other function which is the elemwise_xxx. If we do want an interface for user to call logical_xxx without the broadcast, we can do it in Python just like how add/subtract/multiply/divide is implemented.

Reply:
Isn't add/sub etc.. also using broadcast_add, broadcast_sub ... http://mxnet.incubator.apache.org/api/python/symbol/symbol.html#mathematical-functions ?

Reply:
@anirudhacharya Please take a look at: https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/ndarray/ndarray.py#L2479

Reply:
I think there should be a non broadcast version

Reply:
Will add it.